test/cql-pytest/test_alter_table.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+# Tests for alter table statement
+
+import time
+import pytest
+
+from util import new_test_table, unique_name
+from nodetool import flush
+
+# Test checks only case of preparing `ALTER TABLE ... DROP ... USING TIMESTAMP ?` statement.
+# It is `scylla_only` because Cassandra doesn't allow to prepare the statement with ? as the timestamp.
+# More tests about `ALTER TABLE ... DROP ...` are in cassandra_tests/validation/operation/alter_test.py
+def testDropColumnWithTimestampPrepared(scylla_only, cql, test_keyspace):
+    with new_test_table(cql, test_keyspace, "id int, c1 int, v1 int, todrop int, PRIMARY KEY (id, c1)") as table:
+        prepared = cql.prepare(f"ALTER TABLE {table} DROP todrop USING TIMESTAMP ?")
+        for i in range(5):
+            cql.execute(f"INSERT INTO {table} (id, c1, v1, todrop) VALUES (1, {i}, {i}, {i}) USING TIMESTAMP {10000 * i}")
+
+        # Because the tests are run in parallel, we cannot be sure the table will stay in memory.
+        # It's safer to flush the table before dropping the column
+        flush(cql, table)
+        cql.execute(prepared, [20000])
+        cql.execute(f"ALTER TABLE {table} ADD todrop int")
+        cql.execute(f"INSERT INTO {table} (id, c1, v1, todrop) VALUES (1, 100, 100, 100) USING TIMESTAMP 30000")
+        
+        result = set(cql.execute(f"SELECT id, c1, v1, todrop FROM {table}"))
+        assert result == set([
+                (1, 0, 0, None),
+                (1, 1, 1, None),
+                (1, 2, 2, None),
+                (1, 3, 3, 3),
+                (1, 4, 4, 4),
+                (1, 100, 100, 100)
+        ])
+
+# We allow to prepare alter table statement when the schema doesn't exist yet
+def testDropColumnWithTimestampPreparedNonExistingSchema(scylla_only, cql, test_keyspace):
+    table = test_keyspace + "." + unique_name()
+    
+    prepared = cql.prepare(f"ALTER TABLE {table} DROP todrop USING TIMESTAMP ?")
```

Comment:
Idea for another test: prepare when `{table}` exists but `todrop` column doesn't and is only added later.

Reply:
Another idea, but not sure whether it would make sense (i.e. maybe the same logic is already tested by the old tests): prepare a statement that does not have a bind marker but rather has an internal constant for the timestamp in the string.

Reply:
The previous patch contains such tests.

Reply:
> The previous patch contains such tests.

Does it? The previous patch un-xfails some existing tests that execute _unprepared_ statements.

Reply:
> The only reason to support bind variables is that Cassandra happens to support them (and also orthogonality is a good thing).

Actually, in this particular case (`ALTER TABLE DROP COLUMN USING TIMESTAMP ?`) Cassandra does _not_ support the bind marker, just checked this.

Reply:
Oh, so we did it for naught.

The grammar does support it, but it's not surprising that the actual code doesn't.